docs/reference/user-defined-functions.md

```diff
@@ -33,6 +33,7 @@ values.
 | `ARRAY`   | `java.util.List`                      |
 | `MAP`     | `java.util.Map`                       |
 | `STRUCT`  | `org.apache.kafka.connect.data.Struct`|
+| `BYTES`   | `java.nio.ByteBuffer`                 |
```

Comment:
I think we'll have to make changes to pass a `bytep[]` to UDF functions and not the ByteBuffer because ByteBuffer can mutate  and users should not be allowed to mutate parameters.

Reply:
I tried to switch from ByteBuffer to either byte[] or my own KsqlBytes (immutable), but I had other issues. One is that UDF support the `Struct` class which allows bytes fields with ByteBuffer or byte[] only. That's my first limitation so I stop switching to KsqlBytes. The second is that byte[] is still mutable. I think we can continue by making ByteBuffer readonly instead of using byte[]. So +1 to this doc.

Reply:
Okay. If we ever decide to switch it to `byte[]` we can update the docs.